mobile_verifier/src/coverage.rs

```diff
@@ -291,6 +334,72 @@ impl CoveredHexes {
     }
 }
 
+pub struct CoveredHexCache {
+    pool: Pool<Postgres>,
+    covered_hexes: Arc<Cache<Uuid, CachedCoverage>>,
+}
+
+impl CoveredHexCache {
+    pub fn new(pool: &Pool<Postgres>) -> Self {
+        let cache = Arc::new(Cache::new());
+        let cache_clone = cache.clone();
+        tokio::spawn(async move {
+            cache_clone
+                .monitor(4, 0.25, std::time::Duration::from_secs(60 * 60 * 24 * 2))
+                .await
+        });
+
+        Self {
+            covered_hexes: cache,
+            pool: pool.clone(),
+        }
+    }
+
+    pub async fn fetch_coverage(&self, uuid: &Uuid) -> Result<Option<CachedCoverage>, sqlx::Error> {
+        if let Some(covered_hexes) = self.covered_hexes.get(uuid).await {
+            return Ok(Some(covered_hexes.clone()));
+        }
+        let Some(cbsd_id) = sqlx::query_scalar("SELECT cbsd_id FROM coverage_claim_time WHERE uuid = $1")
+            .bind(uuid)
+            .fetch_optional(&self.pool)
+            .await? else {
+                return Ok(None);
+            };
+        let coverage: Vec<_> = sqlx::query_as("SELECT * FROM hex_coverage WHERE uuid = $1")
+            .bind(uuid)
+            .fetch_all(&self.pool)
+            .await?
+            .into_iter()
+            .map(|HexCoverage { hex, .. }| CellIndex::try_from(hex as u64).unwrap())
+            .collect();
+        let cached_coverage = CachedCoverage { cbsd_id, coverage };
+        let _ = self
+            .covered_hexes
+            .insert(
+                *uuid,
+                cached_coverage.clone(),
+                std::time::Duration::from_secs(60 * 60 * 24),
+            )
+            .await;
+        Ok(Some(cached_coverage))
+    }
+}
+
+#[derive(Clone)]
+pub struct CachedCoverage {
+    pub cbsd_id: String,
+    coverage: Vec<CellIndex>,
+}
+
+impl CachedCoverage {
+    pub fn max_distance_km(&self, latlng: LatLng) -> f64 {
```

Comment:
would it be more accurate or more efficient to do this distance calculation with the native h3 grid_distance functions (`grid_disk_distances_fast/safe()`)? could we confidently say due to RF propagation that coverage has to be within N range of the k-ring from the "source", get the `(CellIndex, u32)` iterator and return the furthest of those that's also a member of the `coverage` vec?